### PR TITLE
Membership Saves - Create Reminder screen

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What does this change?

Adds a reminder screen and API call post cancellation in the Membership cancellation saves journey. This PR also adds in the API call to send the reason on the previous screen. Also changes the existing reminder component to add a new button.

## How to test

Testing locally:
Enable feature switch and confirm a cancellation on membership save journey. Click `Set Reminder`. The `api/reminders` call does not seem to work locally at time of writing.

## Images

Validation on reasons screen
image

New screen
image

After submission
image

Existing reminder screen
image

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

**Comment by andrewHEguardian** [comment]
[Membership Saves - Set a Reminder[Initial design]](https://trello.com/c/BLUGnMqQ/4094-membership-saves-set-a-reminderinitial-design)

File: cypress/integration/parallel-2/membershipSave.spec.ts [hunk 1]
```diff
@@ -9,6 +9,18 @@ import { signInAndAcceptCookies } from '../../lib/signInAndAcceptCookies';
 
 if (featureSwitches.membershipSave) {
 	describe('Cancel membership saves', () => {
+		const setSignInStatus = () => {
+			cy.window().then((window) => {
+				// @ts-ignore
```
Comment (charleycampbell) [inline cypress/integration/parallel-2/membershipSave.spec.ts:14]: Do we need this comment/ts-ignore?

**Review by charleycampbell — APPROVED** [review]
Looks good

File: cypress/integration/parallel-2/membershipSave.spec.ts (hunk 1 above)
Comment (andrewHEguardian) [reply cypress/integration/parallel-2/membershipSave.spec.ts:14]: I think we could add the same config from `shared/globals.ts` to `cypress/support/index.ts` (or similar) but I don't know if it's worth it

**Review by charleycampbell — COMMENTED** [review]
Looks good just a couple of questions 👍

File: client/components/mma/cancel/cancellationSaves/CancellationSaves.stories.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { MembershipCancellationLanding } from './MembershipCancellationLanding';
 import { MembershipSwitch } from './MembershipSwitch';
 import { SaveOptions } from './SaveOptions';
 import { SelectReason } from './SelectReason';
-import { SwitchThankYou } from './SwitchThankYou';
```
[the remark this answers is not in the document]
Comment (andrewHEguardian) [reply client/components/mma/cancel/cancellationSaves/CancellationSaves.stories.tsx:16]: Hmm. Looks like the merge conflict didn't go as expected, yes it should still be there

File: cypress/integration/parallel-2/membershipSave.spec.ts [hunk 3]
```diff
@@ -9,6 +9,17 @@ import { signInAndAcceptCookies } from '../../lib/signInAndAcceptCookies';
 
 if (featureSwitches.membershipSave) {
 	describe('Cancel membership saves', () => {
+		const setSignInStatus = () => {
+			cy.window().then((window) => {
+				window.guardian.identityDetails = {
+					signInStatus: 'signedInRecently',
+					userId: '200006712',
```
[the remark this answers is not in the document]
Comment (andrewHEguardian) [reply cypress/integration/parallel-2/membershipSave.spec.ts:16]: I'm sure it is, and this is just copied from the other tests, but happy to change them to something more obviously test data

**Review by charleycampbell — DISMISSED** [review]
Looks great!

**Pull request merged**